Runtime/Interaction/SnapZone.cs

```diff
@@ -317,27 +337,46 @@ public override void ProcessInteractor(XRInteractionUpdateOrder.UpdatePhase upda
             {
                 base.ProcessInteractor(updatePhase);
             }
-
+            
             if (socketActive)
             {
-                if (m_HoverTargets.Count == 0 && ShowHighlightObject)
+                hoverTargets.RemoveAll(target => target == null || target.enabled == false);
```

Comment:
This can be moved to The first if in `CheckForReleasedHoverTargets` (L355) since it is only relevant there and there is no need to go through the whole list potentially twice in a method that is called several times per frame.